finality-aleph/src/new_network/mod.rs

```diff
@@ -85,18 +87,24 @@ impl Protocol {
 
 type NetworkEventStream = Pin<Box<dyn Stream<Item = Event> + Send>>;
 
+#[derive(Copy, Clone)]
+pub enum NetworkSendError {
+    Closed,
+}
+
 /// Abstraction over a network.
+#[async_trait]
 pub trait Network: Clone + Send + Sync + 'static {
     /// Returns a stream of events representing what happens on the network.
     fn event_stream(&self) -> NetworkEventStream;
 
-    /// A sender for sending messages to the given peer, see the substrate network docs for how to
-    /// use it. Returns None if not connected to the peer.
-    fn message_sender(
-        &self,
+    /// A method for sending data to the given peer using a given protocol. Returns Error if not connected to the peer.
+    async fn send<'a>(
```

Comment:
Unfortunately because of method `ready<'a>` in `NotificationSender` depending on lifetime and returning `NotificationSenderReady<'a>` we weren't able to find any other solution with Timorl, to provide a mockable interface for Network (the issue is in creating a trait for `NotificationSender` and `NotificationSenderReady`). It was because of this lifetime and `NotificationSenderReady::send(...)` using `self` instead of reference.